src/utils/translation.cpp

```diff
@@ -283,26 +283,22 @@ const wchar_t* Translations::fribidize(const wchar_t* in_ptr)
 #if ENABLE_BIDI
     if(this->isRTLLanguage())
     {
-        const int FRIBIDI_BUFFER_SIZE = 512;
-        FriBidiChar fribidiInput[FRIBIDI_BUFFER_SIZE];
-        int len = 0;
-        int n = 0;
-        //std::cout << "fribidi input : ";
-        for (n = 0; ; n++)
-        {
-            fribidiInput[n] = in_ptr[n];
-            //std::cout << (int)fribidiInput[n] << " ";
-            len++;
+        std::size_t length = wcslen(in_ptr);
+        FriBidiChar *fribidiInput;
 
-            if (n == FRIBIDI_BUFFER_SIZE-1) // prevent buffeoverflows
-            {
-                Log::warn("Translations::fribidize", "translated string too long, truncating");
-                fribidiInput[n] = 0;
-                break;
-            }
-            if (fribidiInput[n] == 0) break; // stop on '\0'
+        if (sizeof(wchar_t) == sizeof(FriBidiChar))
+            fribidiInput = (FriBidiChar*) in_ptr;
```

Comment:
I am not sure that this is right. I believe fribidiInput will be modified, and we certainly don't want to modify the original string while translating, so pretty sure we do need a copy